controller/authController.ts

```diff
@@ -0,0 +1,25 @@
+import type {Request, Response, Application, response} from "express";
```

Comment:
Missing unit tests for this file

Reply:
How do i unit test a controller?

Reply:
I would split the code into two:

1) Routes which just contain 'app.get' and 'app.post'
2) Functions which perform the logic on the route.

For example:

router.ts (imported into the app.ts
```
const router = express.Router();

router.get('/login', LoginController.get);
router.post('/login', LoginController.potst);

export default router
```

app.ts
```
  app.use('/', router);
```

login controller (You could use functions instead of classes and export them)
```
export class LoginController {

    public get(req:Request, res:Response): void {
        res.render("login")
    }

   public post((req:Request, res:Response): void {
        let data: Login = req.body;

        try{
            req.session.token  = await login(data)
            console.log(req.session)
            res.redirect('/index')
        }
        catch(e){
            console.error(e)
            res.locals.errormessage = (e as Error).message
            res.render('login', req.body)
        }
    }


}
```

You then have a way of unit testing the controller code, as you can mock out the req/res objects you pass in and any services that may be called